am-lib/src/main/java/uk/gov/hmcts/reform/amlib/DefaultRoleSetupImportService.java

```diff
@@ -0,0 +1,216 @@
+                                      @NonNull String resourceType,
+                                      @NonNull String resourceName) {
+        if (serviceName.isEmpty()) {
+            throw new IllegalArgumentException(emptyServiceError);
+        }
+
+        if (resourceType.isEmpty() || resourceName.isEmpty()) {
+            throw new IllegalArgumentException("Resource cannot contain empty values");
+        }
+
+        jdbi.useExtension(DefaultRoleSetupRepository.class, dao ->
+            dao.addResourceDefinition(serviceName, resourceType, resourceName));
+
+    }
+
+    /**
+     * Creates a new resource attribute with default permissions for a role or updates if already exists.
+     *
+     * <p>Operation uses a transaction and will rollback if any errors are encountered whilst adding entries.
+     * {@link PersistenceException}
+     *
+     * @param defaultPermissionGrant a container for granting default permissions.
+     */
+    public void grantDefaultPermission(DefaultPermissionGrant defaultPermissionGrant) {
+        jdbi.useTransaction(handle -> {
+            DefaultRoleSetupRepository dao = handle.attach(DefaultRoleSetupRepository.class);
+            try {
+                defaultPermissionGrant.getAttributePermissions().forEach((attribute, permissionAndClassification) -> {
+                    dao.createResourceAttribute(ResourceAttribute.builder()
+                        .serviceName(defaultPermissionGrant.getServiceName())
+                        .resourceName(defaultPermissionGrant.getResourceName())
+                        .resourceType(defaultPermissionGrant.getResourceType())
+                        .attribute(attribute.toString())
+                        .securityClassification(permissionAndClassification.getValue())
+                        .build()
+                    );
+
+                    dao.grantDefaultPermission(
+                        DefaultPermission.builder()
+                            .serviceName(defaultPermissionGrant.getServiceName())
+                            .resourceType(defaultPermissionGrant.getResourceType())
+                            .resourceName(defaultPermissionGrant.getResourceName())
+                            .attribute(attribute.toString())
+                            .roleName(defaultPermissionGrant.getRoleName())
+                            .permissions(Permissions.sumOf(permissionAndClassification.getKey()))
+                            .build());
+                });
+
+            } catch (Exception e) {
+                throw new PersistenceException(e);
+            }
+        });
+    }
+
+    /**
+     * Deletes all default permissions within a service for a given resource type.
+     *
+     * <p>Operation uses a transaction and will rollback if any errors are encountered whilst adding entries.
+     *
+     * @param serviceName  the name of the service to delete default permissions for.
```

Comment:
Dot's here as well. They are OK in the first / main section but not when used with @param or @returns or @throws.